Makefile

```diff
@@ -128,16 +129,16 @@ devtools-stamp: tools
 devtools-clean: tools-clean
 	rm -f devtools-stamp
 
-vendor-deps: tools
-	@echo "--> Generating vendor directory via dep ensure"
-	@rm -rf .vendor-new
-	@dep ensure -v -vendor-only
-	tar -c vendor/ | sha1sum | cut -d' ' -f1 > $@
+go-mod-cache: go.sum
+	@echo "--> Download go modules to local cache"
+	@go mod download
+
+go.sum: go.mod
+	@echo "--> Generating vendor directory via go mod vendor"
+	@go mod vendor
 
-update_vendor_deps: tools
-	@echo "--> Running dep ensure"
-	@rm -rf .vendor-new
-	@dep ensure -v
+vendor-deps: go.sum
+	go run ./cmd/gosum/main.go go.sum > $@
```

Comment:
Aren't the hashes checked via the `go` command (via the `go.sum` file)? Seems so:
> The go.mod files in a build provide enough information for 100% reproducible builds.

> If someone clones your repository and downloads your dependencies using the go command, they will receive an error if there is any mismatch between their downloaded copies of your dependencies and the corresponding entries in your go.sum.

https://github.com/golang/go/wiki/Modules#should-i-commit-my-gosum-file-as-well-as-my-gomod-file

Reply:
We wanted to include the hash of the dependencies set in the `version` command.

Reply:
I'm still not sure I understand what you guys are doing here. Hashing go.sum does not provide any guarantees. If anyone wanted to verify a build matches, they would still want to verify go.sum / go.mod via the go tool instead of hashing the go.sum file. But maybe I'm missing some context here?